src/microbe_stage/Membrane.cs

```diff
@@ -411,52 +388,40 @@ private void InitializeMesh()
             }
         }
 
-        previousWorkBuffer.Capacity = vertices2D.Capacity;
-        nextWorkBuffer.Capacity = previousWorkBuffer.Capacity;
+        // Make the length longer to guarantee that everything fits easily inside the square
+        cellDimensions *= 100;
+
+        var startingBuffer = new List<Vector2>();
 
         for (int i = membraneResolution; i > 0; i--)
         {
-            previousWorkBuffer.Add(new Vector2(-cellDimensions,
+            startingBuffer.Add(new Vector2(-cellDimensions,
                 cellDimensions - 2 * cellDimensions / membraneResolution * i));
         }
 
         for (int i = membraneResolution; i > 0; i--)
         {
-            previousWorkBuffer.Add(new Vector2(
+            startingBuffer.Add(new Vector2(
                 cellDimensions - 2 * cellDimensions / membraneResolution * i,
                 cellDimensions));
         }
 
         for (int i = membraneResolution; i > 0; i--)
         {
-            previousWorkBuffer.Add(new Vector2(cellDimensions,
+            startingBuffer.Add(new Vector2(cellDimensions,
                 -cellDimensions + 2 * cellDimensions / membraneResolution * i));
         }
 
         for (int i = membraneResolution; i > 0; i--)
         {
-            previousWorkBuffer.Add(new Vector2(
+            startingBuffer.Add(new Vector2(
                 -cellDimensions + 2 * cellDimensions / membraneResolution * i,
                 -cellDimensions));
         }
 
-        // This needs to actually run a bunch of times as the points moving towards the organelles is iterative.
-        // We use rotating work buffers to save time on skipping useless copies
-        for (int i = 0; i < 40 * cellDimensions; i++)
-        {
-            DrawCorrectMembrane(cellDimensions, previousWorkBuffer, nextWorkBuffer);
-
-            (previousWorkBuffer, nextWorkBuffer) = (nextWorkBuffer, previousWorkBuffer);
-        }
-
-        // Copy final vertex data from the work buffer
+        // Get new membrane points
         vertices2D.Clear();
-
-        // The work buffer not being pointed to as the next, is the one we should read the result from
-        vertices2D.AddRange(previousWorkBuffer);
-
-        previousWorkBuffer.Clear();
-        nextWorkBuffer.Clear();
+        vertices2D.AddRange(GenerateMembranePoints(startingBuffer));
```

Comment:
Could you refactor this as well so that GenerateMembranePoints takes in the list and adds the results directly, so that it can avoid having to allocate another temporary list?